business/checkers/services/port_mapping_checker.go

```diff
@@ -28,6 +28,11 @@ func (p PortMappingChecker) Check() ([]*models.IstioCheck, bool) {
 		for portIndex, sp := range p.Service.Spec.Ports {
 			if strings.ToLower(string(sp.Protocol)) == "udp" {
 				continue
+			} else if sp.AppProtocol != nil {
+				if !kubernetes.MatchPortAppProtocolWithValidProtocols(sp.AppProtocol) {
```

Comment:
Nitpick here. You might want to join this if with the previous one since you already check the nil within the `MatchPortAppProtocol` method.

Reply:
if I join these 2 'if', then 'else' will not be clear whether appProtocol is nil, empty or exist but with wrong format.